lib/hci-socket/gap.js

```diff
@@ -130,7 +130,11 @@ Gap.prototype.onHciLeAdvertisingReport = function(status, type, address, address
       case 0x02: // Incomplete List of 16-bit Service Class UUID
       case 0x03: // Complete List of 16-bit Service Class UUIDs
         for (j = 0; j < bytes.length; j += 2) {
-          serviceUuid = bytes.readUInt16LE(j).toString(16);
+          try {
+            serviceUuid = bytes.readUInt16LE(j).toString(16);
+          } catch(exc) {
+            debug("Error reading 16-bit service class uuid:", exc);
+          }
```

Comment:
This is effectively swallowing the error, right?

Is there no way to salvage the actual payload from the advertisement?

Reply:
No way that I can think of.